tests/test_merge.py

```diff
@@ -277,7 +277,7 @@
         ),
         param(
             (
-                DictConfig({"user007": None}, element_type=User),
+                DictConfig({"user007": "???"}, element_type=User),
```

Comment:
The test is called "merging into None" => maybe it's the `element_type` that should be `Optional[User]` rather than replacing `None` with missing?

Reply:
Good point.

Reply:
This relates to https://github.com/omry/omegaconf/issues/460, which I'll be working on in the near future :)

Reply:
If this doesn't work yet, maybe you can hack around it by generating the DictConfig for this test in a separate function, that would set the `element_type` only after the content has been set?
(keeping a note to change it when fixing #460)

Reply:
After playing with it for a bit I found that the test does not pass even if `element_type` is set to `Optional[User]` manually. This is because `merge` complains about about an optional element type.

Anyway, I've reverted the test to it's original state in https://github.com/omry/omegaconf/pull/749/commits/007a879cca593bf5c6351ef33e0f0f15187fd86f (expecting a CI failure).

Reply:
Since #579 and #460 have similar scope (both dealing with optional/non-optional `element_type` in containers), I decided to tackle them both in this PR.